PULL_REQUEST_WORKFLOW.rst

```diff
@@ -0,0 +1,260 @@
+   relevant airflow sources (airflow, chart, tests, kubernetes_tests) first and then we count how many
+   files changed in different packages:
+
+   a) in any case tests in ``Always`` folder are run. Those are special tests that should be run any time
+      modifications to any Python code occurs. Example test of this type is verifying proper structure of
+      the project including proper naming of all files.
+   b) if any of the Airflow API files changed we enable ``API`` test type
+   c) if any of the Airflow CLI files changed we enable ``CLI`` test type
+   d) if any of the Provider files changed we enable ``Providers`` test type
+   e) if any of the WWW files changed we enable ``WWW`` test type
+   f) if any of the Kubernetes files changed we enable ``Kubernetes`` test type
+   g) Then we subtract count of all the ``specific`` above per-type changed files from the count of
+      all changed files. In case there are any files changed, then we assume that some unknown files
+      changed (likely from the core of airflow) and in this case we enable all test types above and the
+      Core test types - simply because we do not want to risk to miss anything.
+   h) In all cases where tests are enabled we also add Heisentests, Integration and - depending on
+      the backend used = Postgres or MySQL types of tests.
+
+9) Quarantined tests are always run when tests are run - we need to run them often to observe how
+   often they fail so that we can decide to move them out of quarantine. Details about the
+   Quarantined tests are described in `TESTING.rst <TESTING.rst>`_
+
+10) There is a special case of static checks. In case the above logic determines that the CI image
+    needs to be build, we run long and more comprehensive version of static checks - including Pylint,
+    MyPy, Flake8. And those tests are run on all files, no matter how many files changed.
+    In case the image is not built, we run only simpler set of changes - the longer static checks
+    that require CI image are skipped, and we only run the tests on the files that changed in the incoming
+    commit - unlike pylint/flake8/mypy, those static checks are per-file based and they should not miss any
+    important change.
+
+Similarly to selective tests we also run selective security scans. In Pull requests,
+the Python scan will only run when there is a python code change and JavaScript scan will only run if
+there is a JavaScript or yarn.lock file change. For master builds, all scans are always executed.
+
+The selective check algorithm is shown here:
+
+.. image:: images/pr/selective_checks.png
+    :align: center
+    :alt: Selective check algorithm
+
+Approval Workflow and Matrix tests
+----------------------------------
+
+As explained above the approval and matrix tests workflow works according to the algorithm below:
+
+1) In case of "doc-only" changes - so changes that do not change any of the code or environment of
```

Comment:
We shouldn't call this "doc-only", as it doesn't apply if someone makes a change in docs/